plugin/db/util/driverutil.go

```diff
@@ -742,3 +705,55 @@ func ConvertYesNo(s string) (bool, error) {
 		return false, errors.Errorf("unrecognized isNullable type %q", s)
 	}
 }
+
+func readRowsForClickhouse(rows *sql.Rows, columnTypes []*sql.ColumnType, columnTypeNames []string, fieldList []db.SensitiveField) ([]interface{}, error) {
+	data := []interface{}{}
+
+	for rows.Next() {
+		cols := make([]interface{}, len(columnTypes))
+		for i, v := range columnTypeNames {
+			// It's a deficiency of clickhouse driver that we can't use sql.NullString to scan nullable fields
+			// as described in https://github.com/ClickHouse/clickhouse-go/issues/754
+			// We utilize reflect for nullable fields and sql.NullString for others.
+			// TODO: remove this workaround once fixed.
+			if strings.HasPrefix(v, "NULLABLE") {
+				cols[i] = reflect.New(columnTypes[i].ScanType()).Interface()
+			} else {
+				cols[i] = new(sql.NullString)
+			}
+		}
```

Comment:
Why do we only deal with the `String` type?

Reply:
I thought I could avoid all the type assertion things but I can't...

Reply:
I was hoping that NullString would do the value conversion for me.